virttest/qemu_storage.py

```diff
@@ -127,7 +128,38 @@ def create(self, params, ignore_errors=False):
             qemu_img_cmd = qemu_img_cmd.rstrip(",")
 
             if self.base_tag:
-                qemu_img_cmd += " -b %s" % self.base_image_filename
+                if external_snapshot_driver:
+                    if external_snapshot_driver == "https":
+                        file_ssl_verify = params.get("file_sslverify")
+                        file_readahead = params.get("file_readahead")
+                        file_url = "{}://{}/{}".format(
+                            external_snapshot_driver,
+                            params.get("remote_host"),
+                            os.path.basename(self.base_image_filename))
+                        jdict = '{"file.driver": %s, "file.url": %s, '
+                        jdict += '"file.sslverify": %s, "file.readahead": %s}'
+                        jdict = jdict % ('\"%s\"' % external_snapshot_driver,
+                                         '\"%s\"' % file_url,
+                                         '\"%s\"' % file_ssl_verify,
+                                         '\"%s\"' % file_readahead)
+                        jdict = "\'json: {}\'".format(jdict)
+                        qemu_img_cmd += " -b %s" % jdict
+                    elif external_snapshot_driver == "ssh":
+                        file_host = params.get("remote_host")
```

Comment:
now in our test, we just deploy http server in localhost ,maybe can deploy it in other host, so name it remote_host.